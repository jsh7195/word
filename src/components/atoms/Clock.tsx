import React, { useEffect, useState } from "react";
import { ClockType } from '@lib/data';
import { ClockStyle } from '@atoms/style';

const Clock = (props: ClockType):JSX.Element => {
  const { limit, text } = props;
  const [ time , setTime ] = useState<number>(0);

  //동일한 제한시간을 가진데이터가 존재해서 index를 붙임
  useEffect(() => {
    setTime(props.limit);
  }, [(props.index+props.limit)])

  useEffect(() => {
    
    const timer = setTimeout(()=>{ setTime(time-1) },1000);
    if( time<-1 ) clearTimeout(timer);
    return () => {
      clearTimeout(timer);
    };
  }, [text,time]);

  return (
    <ClockStyle>
      <span id="timer">남은 시간: {time} </span> / {limit}
    </ClockStyle>
  );
};

export default Clock;
