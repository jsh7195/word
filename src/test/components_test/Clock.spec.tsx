import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from 'react-test-renderer';
import Clock from "@atoms/Clock";

Enzyme.configure({ adapter: new Adapter() });

test("Clock Render Test ... <Clock/>", () => {
  const props = {
    limit: 9,
    text: "test",
    index: 1,
  };
  const component = renderer.create(<Clock {...props} />).toJSON();
  expect(component).toMatchSnapshot();
});
