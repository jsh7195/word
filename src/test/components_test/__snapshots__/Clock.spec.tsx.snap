// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clock Render Test ... <Clock/> 1`] = `
<div>
  <span
    id="timer"
  >
    0
     
  </span>
   / 
  9
</div>
`;
